datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -422,4 +439,76 @@ mod tests {
         ])) as ArrayRef;
         assert_eq!(&output, &array);
     }
+
+    #[test]
+    fn test_nullable_primitive_equal_to() {
```

Comment:
I wonder if there is some way to write a reproducer in an end to end test (as in .slt as well) 🤔

Reply:
🤔 I am trying it, but I am not sure if we can reproduce it with a tiny dataset.
I found it through the fuzz tests in #12667 . And to be honest, I am still confused about why the null row will have the non-default value...

Reply:
I got it.

It may be due to the `compute::take` function using to generate the random dataset.
https://github.com/Rachelint/arrow-datafusion/blob/9ad971be0c6c808e77a74cdfc571a33732a0838a/test-utils/src/array_gen/primitive.rs#L48-L62

And `take`'s implementation:
```
// In `take_primitive`:
let values_buf = take_native(values.values(), indices);
let nulls = take_nulls(values.nulls(), indices);

// In `take_native`:
    match indices.nulls().filter(|n| n.null_count() > 0) {
        Some(n) => indices
            .values()
            .iter()
            .enumerate()
            .map(|(idx, index)| match values.get(index.as_usize()) {
                Some(v) => *v,
                None => match n.is_null(idx) {
                    true => T::default(),
                    false => panic!("Out-of-bounds index {index:?}"),
                },
            })
            .collect(),
```

It will still try to take value from `values` ranther than using `default value`, even the row in `indicies` is null...

This logic seems unreasonable actually?

Reply:
It may be a bit hard to test it end to end test.

The `null row with a non-default value` is only possible to exist in some special cases,
like generating through `take` as mentioned above, or as I remember we use it to improve filter performance in avg accumulator.